### DISCUSSION TITLE
Mitmproxy with NordVPN?

### DISCUSSION THREAD

**Discussion by shreyakannan1205** [mitmproxy/mitmproxy#6184] [Q&A]
Hi all, i just have a quick doubt.
Does MITMPROXY work with NordVPN? Mitmproxy flows don't seem to be working with nordVPN. In mitmproxy events, it always conveys "Client Disconnect" and "Server Disconnect" for every event. 

Any guidance/information would be really helpful!
Thanks!

**Comment by ThomasBucaioni** [comment]
Hello,
When the client makes a request for a webpage, it should do it to the proxy, like in this scheme: https://docs.mitmproxy.org/stable/concepts-howmitmproxyworks/
But when you have a VPN, the request goes to their server, which asks the webpage for you, and then sends it back. So the proxy doesn't see it anymore: https://laptrinhx.com/how-does-a-vpn-work-3505049485/
In your case, the client request is not processed by the proxy
